### ISSUE TITLE
iid mnist data set accuracy does not increase

### ISSUE THREAD

**Issue by lipei246** [TsingZ0/PFL-Non-IID#14]
Hello, in the latest code, there is still an extremely slow increase in average test accuracy at this time the required learning rate is 0.2 but the AUC is increasing rapidly. This problem occurs in mnist of iID. I can't understand why, can you make a solution? The experimental parameters are extracted directly from your readme, and only the learning rate is modified to 0.2.
The final accuracy was 23%,AUC was 65%.

**Comment by TsingZ0** [comment]
Hi, @lipei246 

Which algorithm did you use? 

The hyper-parameters have not been tuned in the `auto_train.sh`. You need to tune other hyper-parameters in addition to the learning rate.

**Comment by lipei246** [comment]
FedAvg
I didn't adopt auto__train.sh. This is my execution statement python main.py -data mnist -m cnn -algo FedAvg -gr 1000 -did 1 -go cnn -lr 0.01 -ls 5 -lbs 32.
Looking for forgiveness and guidance, I'm a novice.

**Comment by lipei246** [comment]
<img width="856" alt="image" src="https://user-images.githubusercontent.com/62500787/175794804-57182865-71d4-4407-bc64-1b0c1e43cc80.png">

**Comment by lipei246** [comment]
I really appreciate your patience with me

**Comment by lipei246** [comment]
I changed the data set to a Dirichlet distribution, and the accuracy of the algorithm came up in line with the paper. I think there may be a problem with minist data set segmentation that should be iID. Or maybe it was my own stupidity. I really appreciate the code you shared and your patient response

**Comment by TsingZ0** [comment]
In addition to the Non-IID feature of the data, the number of data samples among clients is also unbalanced on this platform, which is different from the setting in FedAvg's paper. Other experimental details may also differ from the setting in FedAvg's paper. 

Reproducing the same experimental results is not the goal of this platform, as the included algorithms have their own settings in their papers. 

To reproduce the experimental results, please carefully check the experimental settings in their papers and modify the corresponding codes.

**Comment by TsingZ0** [comment]
Hi, @Mangata1 

I checked the code again, and address this issue by setting `shuffle=True` in `./dataset/utils/dataset_utils.py`. 
Besides, I also update other codes in `./dataset/`. Please check it. 

Since I shuffled the data when using Dirichlet distribution in the old code, the printed accuracy was correct. 

Thank you for your help!

Here is an example for MNIST in the IID setting with cnn:

Data generation: `python generate_mnist.py iid - -`
Simulation: `python -u main.py -lr 0.1 -lbs 16 -nc 20 -jr 1 -nb 10 -data mnist -m cnn -algo FedAvg -gr 2000 -did 0`
![image](https://user-images.githubusercontent.com/24207672/175812723-a5be0cbd-c779-4bd4-ab04-79e64e966c81.png)

**Comment by lipei246** [comment]
<img width="1280" alt="image" src="https://user-images.githubusercontent.com/62500787/176175613-78a19156-add0-4eb3-b5af-e40a7359680c.png">

**Comment by lipei246** [comment]
Hope to get your sincere answer, how should I adjust, forgive my clumsiness.

**Comment by TsingZ0** [comment]
What are your hyper-parameter settings?

You can add weight decay like 
`self.optimizer = torch.optim.SGD(self.model.parameters(), lr=self.learning_rate, weight_decay=self.lamda)` in `clientfomo.py'
